tests/unittests/test_datasource/test_azure.py

```diff
@@ -156,8 +156,19 @@ def construct_valid_ovf_env(data=None, pubkeys=None,
 class TestParseNetworkConfig(CiTestCase):
 
     maxDiff = None
+    fallback_config = {
+        'version': 1,
+        'config': [{
+            'type': 'physical', 'name': 'eth0',
+            'mac_address': '00:11:22:33:44:55',
+            'params': {'driver': 'hv_netsvc'},
+            'subnets': [{'type': 'dhcp'}],
+        }]
+    }
 
-    def test_single_ipv4_nic_configuration(self):
+    @mock.patch('cloudinit.sources.DataSourceAzure.device_driver',
```

Comment:
@TheRealFalcon @blackboxsw, in #539, the PR added a netplan driver filter when using the `hv_netvsc` driver.

However, that PR causes these UTs to fail execution on machines where the networking device driver is `hv_netvsc`. I managed to catch this as my machine's networking device driver happens to be `hv_netvsc`. This is due to the `device_driver` not being mocked out in several UTs. I'll be including these required `device_driver` mocks in this UT instead.

Reply:
Good test fixes thanks.